src/index.ts

```diff
@@ -3,64 +3,98 @@ import path from 'path'
 import { homedir } from 'os'
 import deepmerge from 'deepmerge'
 import { extendConfig } from 'hardhat/config'
-import { HardhatConfig, NetworkConfig, NetworksConfig, HardhatUserConfig } from 'hardhat/types'
+import { HardhatConfig, NetworkConfig, NetworksUserConfig, NetworkUserConfig, HardhatUserConfig } from 'hardhat/types'
+import { HardhatPluginError } from 'hardhat/plugins'
+import { parseLocalNetworksConfigPath } from './utils'
 
 const HARDHAT_CONFIG_DIR = '.hardhat'
 const HARDHAT_NETWORK_CONFIG_FILE = 'networks.json'
 
 export interface LocalNetworksConfig {
-  networks: NetworksConfig
-  defaultConfig: NetworkConfig
+  networks?: NetworksUserConfig
+  defaultConfig?: NetworkUserConfig
+}
+
+interface LocalNetworksConfigInternal extends LocalNetworksConfig {
+  networks: NetworksUserConfig
+  defaultConfig: NetworkUserConfig
 }
 
 extendConfig((hardhatConfig: HardhatConfig, userConfig: HardhatUserConfig): void => {
-  const localNetworksConfig = readLocalNetworksConfig(userConfig)
+  const homeLocalNetworksConfig = readHomeNetworksConfig()
```

Comment:
I find `homeLocal` a bit redundant, what do you think about distinguishing between the home config (default) and the local (custom) one?

```suggestion
  const homeNetworksConfig = readHomeNetworksConfig()
```